### ISSUE TITLE
Support entering floats with a comma as decimal separator

### ISSUE THREAD

**Issue by 5H4D0W-X** [Yellow-Dog-Man/Resonite-Issues#1802]
### Is your feature request related to a problem? Please describe.

Some keyboard layouts replace the decimal key on the numpad with a comma key, as that is used as the decimal separator in some languages (1.234.567,89). Currently when a number is entered with a comma, it is ignored (probably to make it possible to input numbers that are separated into 3-digit groups, such as 12,345.67)

### Describe the solution you'd like

I can see two options to solving this:

- an opt-in setting that makes commas behave like dots
- a more adaptive float parser, where only the last separator in the number counts (so that 1,234.56 is parsed without the comma while 1234,56 or 1.234,56 is, and 1,234,56 parses using the second comma as its decimal separator)

### Describe alternatives you've considered

Mods, but there aren't any that do this afaik

### Requesters

ShadowX

**Comment by troyBORG** [comment]
I know when you're changing the language I've seen it use a comma instead of a period like when Frooxius take a time photo.  It would show 5,2 second for the countdown timer

**Comment by 5H4D0W-X** [comment]
It does, but that does not apply to float parsing. That makes it even more problematic because grabbing a value out of a float field would sometimes give you a number formatted with a comma, which you couldn't actually drop anywhere with correct formatting.

**Comment by Banane9** [comment]
I thought Froox fixed this with #905 / #1639 / #1655 🤔

**Comment by 5H4D0W-X** [comment]
He did, that's why I (tried to) use the past tense. But I guess that's besides the point of this issue anyway

**Comment by Banane9** [comment]
Oh, I see - your issue is just about successfully parsing floats that mix comma and points, by using whatever is the last one as the decimal separator?

**Comment by Readun** [comment]
I think its also a huge issue when trying to deal with websockets too. If it wasnt accounted for in flux, people with that syntax change need to modify the flux for themself and vica versa.

**Comment by 5H4D0W-X** [comment]
> Oh, I see - your issue is just about successfully parsing floats that mix comma and points, by using whatever is the last one as the decimal separator?

exactly, I don't see a downside to doing it that way (apart from comma separated lists, but standard formatting uses semicolons and trailing separators don't matter anyway) and it'd be much more forgiving. And as mentioned, if that method is too involved or too risky I'd still appreciate some sort of setting to opt in to
